pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java

```diff
@@ -162,28 +166,44 @@ public static GroupBuildInfo run(
 
         RepositoryData repo = null;
 
-        if (repoZipPath != null) {
-            repo = parseRepository(new File(repoZipPath));
+        if (repoZipPath != null || context.getPigConfiguration()
+                .getFlow()
+                .getRepositoryGeneration()
+                .getStrategy() != RepoGenerationStrategy.IGNORE) {
+            if (repoZipPath != null) {
+                repo = parseRepository(new File(repoZipPath));
+            } else {
+                repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            }
+            context.setRepositoryData(repo);
+            context.storeContext();
         } else {
-            repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            log.info("Skipping Repo Generation");
         }
-        context.setRepositoryData(repo);
-        context.storeContext();
 
-        if (!skipSources) {
+        if (!(skipSources || context().getPigConfiguration()
+                .getFlow()
+                .getSourcesGeneration()
+                .getStrategy() == SourcesGenerationStrategy.IGNORE)) {
             generateSources();
 
         } else {
             log.info("Skipping Source Generation");
         }
 
-        if (!skipJavadoc) {
+        if (!(skipJavadoc || context.getPigConfiguration()
+                .getFlow()
+                .getJavadocGeneration()
+                .getStrategy() == JavadocGenerationStrategy.IGNORE)) {
```

Comment:
Should that be `!= IGNORE`?

Reply:
I think I have it ok, `(if **skip**=true OR **strategy**=true)`**!**(don't do if do else)

Reply:
oh right! I didn't notice the `!` at the begininning.